src/vm/methodtable.cpp

```diff
@@ -2526,6 +2534,8 @@ bool MethodTable::ClassifyEightBytesForNativeStruct(SystemVStructRegisterPassing
     }
     CONTRACTL_END;
 
+    _ASSERTE(isNativeStruct);
```

Comment:
Same feedback here: is there not something stronger that we can assert based on the `MethodTable` being classified rather than a parameter passed in by the caller?